src/client/actions/edit/sectionActions.js

```diff
@@ -73,6 +73,67 @@ export const onChangeSection = (key, value) => {
   }
 }
 
+export const onSplitTextSection = (existingSectionBody, newSectionBody) => {
+  return (dispatch, getState) => {
+    const { edit: { article, sectionIndex } } = getState()
+    // update original section with updated content
+    dispatch(onChangeSection("body", existingSectionBody))
+    dispatch(newSection("text", sectionIndex + 1, { body: newSectionBody }))
+
+    if (!article.published) {
+      debouncedSaveDispatch(dispatch)
+    }
+  }
+}
+
+export const onMergeTextSections = newHtml => {
+  return (dispatch, getState) => {
+    const { edit: { sectionIndex } } = getState()
+    dispatch(onChangeSection("body", newHtml))
+    dispatch(removeSection(sectionIndex - 1))
+  }
+}
+
+export const maybeMergeTextSections = () => {
+  return (dispatch, getState) => {
+    const {
+      edit: {
+        article: { sections },
+        section,
+        sectionIndex
+      },
+    } = getState()
+    const sectionBefore = sections[sectionIndex - 1]
```

Comment:
any chances `sections.length == 0`?

Reply:
There is a check for this in the component where it is called from, but I think you're right that it should be called here, especially since its name includes "maybe"

Reply:
Or a check for if sectionIndex is 0, rather -- because this is called from inside of a section, there will always be at least one item in the array.